### PR TITLE
Make affinity in charts more generic

### PR DESCRIPTION
**Pull request by yannick-roeder**

### CODE REVIEW

File: charts/streams-app/templates/deployment.yaml [hunk 1]
```diff
@@ -60,50 +60,23 @@ spec:
...
-        {{- end }}
-        podAntiAffinity:
-        {{- if .Values.requirePodAntiAffinity }}
-          requiredDuringSchedulingIgnoredDuringExecution:
-          - labelSelector:
-            matchExpressions:
-            - key: app
-              operator: In
-              values:
-              - {{ template "streams-app.name" . }}
```
Comment (philipp94831) [inline charts/streams-app/templates/deployment.yaml:79]: Since this is a dynamic value injected by helm, I don't see how we can replicate that with the new way

File: charts/producer-app-cleanup-job/templates/job.yaml [hunk 2]
```diff
@@ -39,24 +38,19 @@ spec:
       serviceAccountName: {{ .Values.serviceAccountName }}
       {{- end }}
       {{- if .Values.tolerations }}
-      tolerations:
-{{ toYaml .Values.tolerations | indent 8 }}
+      tolerations: {{ toYaml .Values.tolerations | nindent 8 }}
```
Comment (philipp94831) [inline charts/producer-app-cleanup-job/templates/job.yaml:41]: Can you do these changes in a separate PR please?

**Review by philipp94831 — CHANGES_REQUESTED** [review]

File: charts/streams-app/templates/deployment.yaml (hunk 1 above)
Comment (torbsto) [reply charts/streams-app/templates/deployment.yaml:79]: You can achieve the same if we use `tpl` in the deployment.yaml, e.g.:
```yaml
      {{- if .Values.affinity }}
      affinity: 
        {{- tpl (toYaml .Values.affinity) . | nindent 8 }}
      {{- end }}
```

and the values.yaml:
```yaml
  podAntiAffinity:
     preferredDuringSchedulingIgnoredDuringExecution:
      - weight: 1
        podAffinityTerm:
          topologyKey: "kubernetes.io/hostname"
          labelSelector:
            matchExpressions:
              - key: app
                operator: In
                values:
                  - '{{ .Values.nameOverride }}'
```

**Review by philipp94831 — CHANGES_REQUESTED** [review]
Please update READMEs

**Review by philipp94831 — APPROVED** [review]
